contracts/Zap.sol

```diff
@@ -0,0 +1,247 @@
+
+    IERC20 public constant ren = IERC20(0xEB4C2781e4ebA804CE9a9803C67d0893436bB27D);
+    IERC20 public constant wbtc = IERC20(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+
+    struct Pool {
+        IERC20 lpToken;
+        ICurveFi deposit;
+        ISett sett;
+    }
+    Pool[4] public pools;
+
+    constructor() public {
+        pools[0] = Pool({ // crvRenWBTC [ ren, wbtc ]
+            lpToken: IERC20(0x49849C98ae39Fff122806C06791Fa73784FB3675),
+            deposit: ICurveFi(0x93054188d876f558f4a66B2EF1d97d16eDf0895B),
+            sett: ISett(0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545)
+        });
+        pools[1] = Pool({ // crvRenWSBTC [ ren, wbtc, sbtc ]
+            lpToken: IERC20(0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3),
+            deposit: ICurveFi(0x7fC77b5c7614E1533320Ea6DDc2Eb61fa00A9714),
+            sett: ISett(0xd04c48A53c111300aD41190D63681ed3dAd998eC)
+        });
+        pools[2] = Pool({ // tbtc-sbtcCrv [ tbtc, ren, wbtc, sbtc ]
+            lpToken: IERC20(0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd),
+            deposit: ICurveFi(0xaa82ca713D94bBA7A89CEAB55314F9EfFEdDc78c),
+            sett: ISett(0xb9D076fDe463dbc9f915E5392F807315Bf940334)
+        });
+        pools[3] = Pool({ // Exclusive to wBTC
+            lpToken: wbtc,
+            deposit: ICurveFi(0x0),
+            sett: ISett(0x4b92d19c11435614CD49Af1b589001b7c08cD4D5) // byvWbtc
+        });
+
+        // Since we don't hold any tokens in this contract, we can optimize gas usage in mint calls by providing infinite approvals
+        for (uint i = 0; i < pools.length; i++) {
+            Pool memory pool = pools[i];
+            pool.lpToken.safeApprove(address(pool.sett), uint(-1));
+            if (i < 3) {
+                ren.safeApprove(address(pool.deposit), uint(-1));
+                wbtc.safeApprove(address(pool.deposit), uint(-1));
+                IERC20(address(pool.sett)).safeApprove(address(settPeak), uint(-1));
+            } else {
+                IERC20(address(pool.sett)).safeApprove(address(byvWbtcPeak), uint(-1));
+            }
+        }
+    }
+
+    /**
+    * @notice Mint ibbtc with wBTC / renBTC
+    * @param token wBTC or renBTC address
+    * @param amount wBTC or renBTC amount
+    * @param poolId 0=crvRenWBTC, 1=crvRenWSBTC, 2=tbtc-sbtcCrv, 3=yvWbtc
+    * @param idx Index of the token in the curve pool while adding liquidity; redundant for yvWbtc
+    * @param minOut Minimum amount of ibbtc to mint. Use for capping slippage while adding liquidity to curve pool.
+    * @return _ibbtc Minted ibbtc amount
+    */
+    function mint(IERC20 token, uint amount, uint poolId, uint idx, uint minOut) external returns(uint _ibbtc) {
+        Pool memory pool = pools[poolId];
+
+        token.safeTransferFrom(msg.sender, address(this), amount);
+
+        if (poolId < 3) { // setts
+            _addLiquidity(pool.deposit, amount, poolId + 2, idx); // pools are such that the #tokens they support is +2 from their poolId.
```

Comment:
nit: this invariant will prob be violated after we add additional pools but I don't think we're doing that anytime soon - maybe a comment or sth to that effect here?

Reply:
Yeah I was thinking, we can deploy a new zap when we add a new pool. Thanks, will add a comment too.